tiledb/sm/rest/curl.cc

```diff
@@ -162,7 +162,7 @@ size_t write_header_callback(
   auto* const header_buffer = static_cast<char*>(res_data);
   auto* const pmHeader = static_cast<HeaderCbData*>(userdata);
 
-  if (pmHeader->uri->empty()) {
+  if (pmHeader->should_cache_redirect && pmHeader->uri->empty()) {
```

Comment:
I don't get this :/ what does this has to do with `should_cache_redirect` ?

Reply:
It was confusing to me as well, but mostly this is explained by [my other comment](https://github.com/TileDB-Inc/TileDB/pull/5429#discussion_r1925983091). It looks like the boolean to disable caching in [Curl::init](https://github.com/tiledb-inc/tiledb/blob/286da187ee9a3bc9d66538ce2ed433cc2c8b759a/tiledb/sm/rest/curl.cc#L241) was never to `false` in practice, and this was an edge case that came up when I disabled it. 

Without checking `pmHeader->should_cache_redirect` here we'll return 0 on line 167 and fail the curl request, even if we have initialized curl with `should_cache_redirect=false` like below

```c++
throw_if_not_ok(curlc.init(
      config_, extra_headers_, &redirect_meta_, &redirect_mtx_, /*should_cache_redirect=*/false));
```

Reply:
Can you add a comment explaining what that updated check does? I am still not sure I get it and I will definitely not get it next time I read this code.